motr/setup.c

```diff
@@ -458,6 +458,7 @@ static void cs_reqh_ctx_fini(struct m0_reqh_context *rctx)
 		m0_free(rctx->rc_services[i]);
 	m0_free(rctx->rc_services);
 	m0_free(rctx->rc_service_fids);
+	m0_free((char*)rctx->rc_addb_stlocation);
```

Comment:
The `rc_addb_stlocation` is allocated here. 
Please change the `strdup()` to `m0_strdup()` so we can catch this allocation if necessary.

```
2277                         M0_STRINGARG('A', "ADDB storage domain location",
2278                                 LAMBDA(void, (const char *s)
2279                                 {
2280                                         char tmp_buf[512];
2281                                         sprintf(tmp_buf, "%s-%d", s, (int)m0_pid());
2282                                         rctx->rc_addb_stlocation = strdup(tmp_buf);
2283                                 })),
```

Reply:
And to avoid possible stack overflow, it's better to replace the above sprintf() with `snprintf()`